test/test_nestedtensor.py

```diff
@@ -18,6 +18,24 @@ def _iter_constructors():
     # yield as_nested_tensor
     yield nested_tensor
 
+# Helper functions to pad a noncontiguous nested tensor
+# can be replaced once to_padded_tensor supports noncontiguous memory
+def noncontiguous_to_padded_tensor(input, shape=None):
+    if shape is None:
+        # for now to_padded_tensor gives the correct sizes
+        # but wrong entries when the buffer memory is noncontiguous
```

Comment:
`to_padded_tensor()` is not fixed here because it is more complicated than I thought. I opened an issue ([#80905](https://github.com/pytorch/pytorch/issues/80985)) to track the update progress.

Currently, for testing purpose, I introduced `noncontiguous_to_padded_tensor` in test/test_nestedtensor.py

Reply:
Ah I see. Thanks for opening the tracking issue! In that case, I'm good with pushing a fix to a future PR, but I'll suggest avoiding a silent correctness issue by loudly erroring for `to_padded_tensor` and `linear` if the NT is non-contiguous.